Tests don't work when Netopeer2 is installed

Hi,
after I do a fresh install of sysrepo, all the tests pass. However, after I install Netopeer2, only the first and the last test passes, and all the others fail. On second try, only the last one passes. Deleting everything in `/dev/shm` doesn't work. Also, if the first test fails, then all the others fail on the "zero connections" check, because the first test leaves the connection behind.
This is the log from the first run:
```
$ ctest --output-on-failure 
Test project /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/build
      Start  1: test_modules
 1/12 Test  #1: test_modules .....................   Passed    2.08 sec
      Start  2: test_validation
 2/12 Test  #2: test_validation ..................Child aborted***Exception:   0.39 sec
[==========] Running 3 test(s).
[INF]: Applying scheduled changes.
[INF]: File "feature-deps.yang" was removed.
[INF]: Module "feature-deps" was removed.
[INF]: File "feature-deps2.yang" was removed.
[INF]: Module "feature-deps2" was removed.
[INF]: Scheduled changes applied.
[INF]: Module "test" scheduled for installation.
[INF]: Module "refs" scheduled for installation.
[INF]: File "test.yang" was installed.
[INF]: Applying scheduled changes.
[INF]: Module "test" will be installed as "refs" module dependency.
[INF]: File "refs.yang" was installed.
[INF]: Module "refs" was installed.
[INF]: Dependency module "test" was installed.
[INF]: Scheduled changes applied.
[INF]: Session 11 (user "vk") created.
[ RUN      ] test_leafref
[INF]: There are no subscribers for changes of the module "test" in running DS.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[ERR]: Leafref "/test:test-leaf" of value "8" points to a non-existing leaf. (path: /refs:lref)
[ERR]: Validation failed.
[ERR]: Leafref "/test:test-leaf" of value "10" points to a non-existing leaf. (path: /refs:lref)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "test" in running DS.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[       OK ] test_leafref
[ RUN      ] test_instid
[ERR]: Required instance of "/refs:l" does not exist. (path: /refs:inst-id)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[ERR]: Required instance of "/refs:ll[.='z']" does not exist. (path: /refs:inst-id)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[ERR]: Required instance of "/refs:lll[key='1']" does not exist. (path: /refs:inst-id)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[ERR]: Required instance of "/test:test-leaf" does not exist. (path: /refs:inst-id)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[INF]: There are no subscribers for changes of the module "test" in running DS.
[ERR]: Required instance of "/test:ll1[.='-3000']" does not exist. (path: /refs:cont/def-inst-id)
[ERR]: Validation failed.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[INF]: There are no subscribers for changes of the module "test" in running DS.
[INF]: There are no subscribers for changes of the module "test" in running DS.
[INF]: There are no subscribers for changes of the module "refs" in running DS.
[       OK ] test_instid
[ RUN      ] test_operational
edit
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_validation.c:236: error: Failure!

      Start  3: test_edit
 3/12 Test  #3: test_edit ........................Child aborted***Exception:   0.29 sec
[==========] Running 14 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_edit.c:54: error: Failure!

      Start  4: test_candidate
 4/12 Test  #4: test_candidate ...................Child aborted***Exception:   0.29 sec
[==========] Running 5 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_candidate.c:54: error: Failure!

      Start  5: test_operational
 5/12 Test  #5: test_operational .................Child aborted***Exception:   0.29 sec
[==========] Running 28 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_operational.c:58: error: Failure!

      Start  6: test_lock
 6/12 Test  #6: test_lock ........................Child aborted***Exception:   0.32 sec
[==========] Running 3 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_lock.c:53: error: Failure!

      Start  7: test_apply_changes
 7/12 Test  #7: test_apply_changes ...............Child aborted***Exception:   0.28 sec
[==========] Running 14 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_apply_changes.c:54: error: Failure!

      Start  8: test_copy_config
 8/12 Test  #8: test_copy_config .................Child aborted***Exception:   0.29 sec
[==========] Running 8 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_copy_config.c:55: error: Failure!

      Start  9: test_rpc_action
 9/12 Test  #9: test_rpc_action ..................Child aborted***Exception:   0.29 sec
[==========] Running 12 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_rpc_action.c:58: error: Failure!

      Start 10: test_notif
10/12 Test #10: test_notif .......................Child aborted***Exception:   0.29 sec
[==========] Running 8 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_notif.c:75: error: Failure!

      Start 11: test_get
11/12 Test #11: test_get .........................Child aborted***Exception:   0.29 sec
[==========] Running 5 test(s).
[ RUN      ] test_cached_datastore
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_get.c:56: error: Failure!

      Start 12: test_process
12/12 Test #12: test_process .....................   Passed    0.38 sec

17% tests passed, 10 tests failed out of 12

Total Test time (real) =   5.48 sec

The following tests FAILED:
	  2 - test_validation (Child aborted)
	  3 - test_edit (Child aborted)
	  4 - test_candidate (Child aborted)
	  5 - test_operational (Child aborted)
	  6 - test_lock (Child aborted)
	  7 - test_apply_changes (Child aborted)
	  8 - test_copy_config (Child aborted)
	  9 - test_rpc_action (Child aborted)
	 10 - test_notif (Child aborted)
	 11 - test_get (Child aborted)
Errors while running CTest
```
This is the second run:
```
$ ctest --output-on-failure 
Test project /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/build
      Start  1: test_modules
 1/12 Test  #1: test_modules .....................Child aborted***Exception:   0.47 sec
[==========] Running 17 test(s).
[ RUN      ] test_install_module
[INF]: Applying scheduled changes.
[INF]: File "ops.yang" was removed.
[INF]: Module "ops" was removed.
[INF]: File "ops-ref.yang" was removed.
[INF]: Module "ops-ref" was removed.
[INF]: Scheduled changes applied.
[INF]: Datastore copied from <startup> to <running>.
[INF]: Module "test-module" scheduled for installation.
[INF]: Applying scheduled changes.
[WRN]: Module "test-module" depends on module "referenced-data", which is not implemented.
[WRN]: Failed to apply some changes, leaving all changes scheduled.
[INF]: Module "test-module" installation unscheduled.
[ERR]: Module "test-module" not scheduled for installation.
[ERR]: Module "test-module" was not found in sysrepo.
[INF]: Module "main-mod" scheduled for installation.
[INF]: Applying scheduled changes.
[INF]: File "main-mod.yang" was installed.
[INF]: Module "main-mod" was installed.
[INF]: Scheduled changes applied.
[INF]: Module "main-mod" scheduled for deletion.
[       OK ] test_install_module
[ RUN      ] test_data_deps
[INF]: Applying scheduled changes.
[INF]: File "main-mod.yang" was removed.
[INF]: Module "main-mod" was removed.
[INF]: Scheduled changes applied.
[ERR]: Module "test" not scheduled for deletion.
[ERR]: Module "test" is already in sysrepo.
0x6 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_modules.c:175: error: Failure!

      Start  2: test_validation
 2/12 Test  #2: test_validation ..................Child aborted***Exception:   0.29 sec
[==========] Running 3 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_validation.c:54: error: Failure!

      Start  3: test_edit
 3/12 Test  #3: test_edit ........................Child aborted***Exception:   0.29 sec
[==========] Running 14 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_edit.c:54: error: Failure!

      Start  4: test_candidate
 4/12 Test  #4: test_candidate ...................Child aborted***Exception:   0.28 sec
[==========] Running 5 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_candidate.c:54: error: Failure!

      Start  5: test_operational
 5/12 Test  #5: test_operational .................Child aborted***Exception:   0.29 sec
[==========] Running 28 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_operational.c:58: error: Failure!

      Start  6: test_lock
 6/12 Test  #6: test_lock ........................Child aborted***Exception:   0.29 sec
[==========] Running 3 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_lock.c:53: error: Failure!

      Start  7: test_apply_changes
 7/12 Test  #7: test_apply_changes ...............Child aborted***Exception:   0.29 sec
[==========] Running 14 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_apply_changes.c:54: error: Failure!

      Start  8: test_copy_config
 8/12 Test  #8: test_copy_config .................Child aborted***Exception:   0.32 sec
[==========] Running 8 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_copy_config.c:55: error: Failure!

      Start  9: test_rpc_action
 9/12 Test  #9: test_rpc_action ..................Child aborted***Exception:   0.30 sec
[==========] Running 12 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_rpc_action.c:58: error: Failure!

      Start 10: test_notif
10/12 Test #10: test_notif .......................Child aborted***Exception:   0.30 sec
[==========] Running 8 test(s).
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_notif.c:75: error: Failure!

      Start 11: test_get
11/12 Test #11: test_get .........................Child aborted***Exception:   0.30 sec
[==========] Running 5 test(s).
[ RUN      ] test_cached_datastore
0x1 != 0
[   LINE   ] --- /home/vk/git/netconf-cli/submodules/dependencies/sysrepo/tests/test_get.c:56: error: Failure!

      Start 12: test_process
12/12 Test #12: test_process .....................   Passed    0.41 sec

8% tests passed, 11 tests failed out of 12

Total Test time (real) =   3.83 sec

The following tests FAILED:
	  1 - test_modules (Child aborted)
	  2 - test_validation (Child aborted)
	  3 - test_edit (Child aborted)
	  4 - test_candidate (Child aborted)
	  5 - test_operational (Child aborted)
	  6 - test_lock (Child aborted)
	  7 - test_apply_changes (Child aborted)
	  8 - test_copy_config (Child aborted)
	  9 - test_rpc_action (Child aborted)
	 10 - test_notif (Child aborted)
	 11 - test_get (Child aborted)
Errors while running CTest
```

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -32,7 +32,12 @@ endforeach(test_name)
 foreach(test_name IN LISTS tests)
     target_link_libraries(${test_name} ${CMOCKA_LIBRARIES} sysrepo)
     add_test(NAME ${test_name} COMMAND $<TARGET_FILE:${test_name}>)
-    set_property(TEST ${test_name} APPEND PROPERTY ENVIRONMENT "MALLOC_CHECK_=3 CMOCKA_TEST_ABORT=1")
+    set_property(TEST ${test_name} APPEND PROPERTY ENVIRONMENT
+        "MALLOC_CHECK_=3"
+        "CMOCKA_TEST_ABORT=1"
+        "SYSREPO_REPOSITORY_PATH=${PROJECT_BINARY_DIR}/test_repositories/${test_name}"
+        "SYSREPO_SHM_PREFIX=_tests_sr_${test_name}"
+        )
 endforeach(test_name)
 
 # measure_performance benchmark binary

